src/server/scripts/BrokenIsles/zone_azsuna.cpp

```diff
@@ -87,8 +87,64 @@ class scene_azsuna_runes : public SceneScript
     }
 };
 
+class questnpc_soul_gem : public CreatureScript
+{
+public:
+    questnpc_soul_gem() : CreatureScript("questnpc_soul_gem") { }
+
+
+
+    struct questnpc_soul_gemAI : public ScriptedAI
+    {
+        questnpc_soul_gemAI(Creature* creature) : ScriptedAI(creature) { }
+        bool check = false;
+
+        virtual void UpdateAI(uint32 diff) override
+        {
+            if(check == false)
+            { 
+            checkfordeaddemons(me);
+            check = true;
```

Comment:
If you want to do it only once, you should use the Reset() method and drop the "check" var :)

Reply:
thx